i2c/bmp280_i2c/bmp280_i2c.c

```diff
@@ -0,0 +1,254 @@
+const uint8_t REG_DIG_T3_LSB = 0x8C;
+const uint8_t REG_DIG_T3_MSB = 0x8D;
+const uint8_t REG_DIG_P1_LSB = 0x8E;
+const uint8_t REG_DIG_P1_MSB = 0x8F;
+const uint8_t REG_DIG_P2_LSB = 0x90;
+const uint8_t REG_DIG_P2_MSB = 0x91;
+const uint8_t REG_DIG_P3_LSB = 0x92;
+const uint8_t REG_DIG_P3_MSB = 0x93;
+const uint8_t REG_DIG_P4_LSB = 0x94;
+const uint8_t REG_DIG_P4_MSB = 0x95;
+const uint8_t REG_DIG_P5_LSB = 0x96;
+const uint8_t REG_DIG_P5_MSB = 0x97;
+const uint8_t REG_DIG_P6_LSB = 0x98;
+const uint8_t REG_DIG_P6_MSB = 0x99;
+const uint8_t REG_DIG_P7_LSB = 0x9A;
+const uint8_t REG_DIG_P7_MSB = 0x9B;
+const uint8_t REG_DIG_P8_LSB = 0x9C;
+const uint8_t REG_DIG_P8_MSB = 0x9D;
+const uint8_t REG_DIG_P9_LSB = 0x9E;
+const uint8_t REG_DIG_P9_MSB = 0x9F;
+
+const uint8_t BMP280_WRITE_MODE = 0xFE;
+const uint8_t BMP280_READ_MODE = 0xFF;
+
+struct bmp280_calib_param {
+  // temperature params
+  uint16_t dig_t1;
+  int16_t dig_t2;
+  int16_t dig_t3;
+
+  // pressure params
+  uint16_t dig_p1;
+  int16_t dig_p2;
+  int16_t dig_p3;
+  int16_t dig_p4;
+  int16_t dig_p5;
+  int16_t dig_p6;
+  int16_t dig_p7;
+  int16_t dig_p8;
+  int16_t dig_p9;
+};
+
+#ifdef i2c_default
+void bmp280_init() {
+  // use the "handheld device dynamic" optimal setting (see datasheet)
+  uint8_t buf[2];
+
+  // 500ms sampling time, x16 filter, not set
+  const uint8_t reg_config_val = ((0x04 << 5) | (0x05 << 2)) & 0xFC;
+
+  // LSB of slave address sets read or write mode
+  // send register number followed by its corresponding value
+  buf[0] = REG_CONFIG;
+  buf[1] = reg_config_val;
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), buf, 2, false);
+
+  // osrs_t x1, osrs_p x4, normal mode operation
+  const uint8_t reg_ctrl_meas_val = (0x01 << 5) | (0x03 << 2) | (0x03);
+  buf[0] = REG_CTRL_MEAS;
+  buf[1] = reg_ctrl_meas_val;
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), buf, 2, false);
+}
+
+void bmp280_read_raw(int32_t* temp, int32_t* pressure) {
+  // BMP280 data registers are auto-incrementing and we have 3 temperature and
+  // pressure registers each, so we start at 0xF7 and read 6 bytes to 0xFC note:
+  // normal mode does not require further ctrl_meas and config register writes
+
+  uint8_t buf[6];
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), &REG_PRESSURE_MSB, 1, true);  // true to keep master control of bus
+  i2c_read_blocking(i2c_default, (addr & BMP280_READ_MODE), buf, 6, false);  // false - finished with bus
+
+  // store the 20 bit read in a 32 bit signed integer for conversion
+  *pressure = (buf[0] << 12) | (buf[1] << 4) | (buf[2] >> 4);
+  *temp = (buf[3] << 12) | (buf[4] << 4) | (buf[5] >> 4);
+}
+
+void bmp280_reset() {
+  // reset the device with the power-on-reset procedure
+  uint8_t buf[2] = { REG_RESET, 0xB6 };
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), buf, 2, false);
+}
+
+int32_t bmp280_convert(int32_t* temp, struct bmp280_calib_param* params) {
```

Comment:
For this and the next 3 functions, why are we passing pointers to int32_t ... these are not in/out vars (I don't think)
If they were to remain pointers, they should be `const` pointers, but please change them to not be const pointers at all.

Note you can try doing
`target_compile_options(bmp280_i2c -Wall)` in your CMakeLists.txt just to see what sort of warnings you may get from the compiler

Reply:
Done